### PR TITLE
feat: CXSPA-5901 Bump eslint-plugin-deprecation and fix related issues

### PR DESCRIPTION
**Pull request by pawelfras**
- bump "eslint-plugin-deprecation" to v2.0.0
- add "@typescript-eslint/utils" v6.13.1 explicitly to our devDependencies as this library is used in eslint-rules library
- change module resolution of tools/eslint-rules to "node16" as it's required for "@typescript-eslint/utils" v6
- add "@swc-node/register" v1.6.8 and "@swc/core" v1.3.85 as it's compiler recommended by Nx team for linting (increase performance + works OOTB with new tools/eslint-rules output type)
- adjust typing in tools/eslint-rules
- adjust eslint rule to work with "@typescript-eslint/utils" v6

closes [CXSPA-5901](https://jira.tools.sap/browse/CXSPA-5901)

### CODE REVIEW

File: tools/eslint-rules/rules/use-default-provide-config.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const rule = ESLintUtils.RuleCreator(() => __filename)({
     type: 'problem',
     docs: {
       description: ``,
-      recommended: 'error',
+      recommended: 'recommended',
```
Comment (Platonn) [inline tools/eslint-rules/rules/use-default-provide-config.ts:31]: Why such a change?
Does it change the severity to lower, e.g. from error to warning/info?

File: tools/eslint-rules/tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
-  "extends": "../../tsconfig.json",
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "module": "commonjs"
+    "module": "Node16",
+    "moduleResolution": "Node16"
```
Comment (Platonn) [inline tools/eslint-rules/tsconfig.json]: Why not Node18, Node20 or even NodeNext?

Comment (pawelfras) [reply tools/eslint-rules/tsconfig.json]: I did it based on what was done in the Nx repo when `@typescript-eslint/utils` v6 was introduced: https://github.com/nrwl/nx/pull/19548/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519 
The next possible value is `NodeNext` and if you wish, we can try it out.

File: tools/eslint-rules/rules/use-default-provide-config.ts (hunk 1 above)
Comment (pawelfras) [reply tools/eslint-rules/rules/use-default-provide-config.ts:31]: Such a change was forced by changes in the API of `@typescript-eslint`: https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/#reworked-configuration-names

The `recommended` value is what we need now. It doesn't affect the severity as we set the level of this rule in `.eslintrc.json` file: https://github.com/SAP/spartacus/blob/57f492f8802b918ea11c48414d87927e9373fe6f/.eslintrc.json#L111

File: tools/eslint-rules/tsconfig.json (hunk 2 above)
Comment (pawelfras) [reply tools/eslint-rules/tsconfig.json]: After tests, value of the `module` and `moduleResolution` has been set to `NodeNext`

**Review by Platonn — APPROVED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**